driver/ppm_events.c

```diff
@@ -738,6 +740,11 @@ int val_to_ring(struct event_filler_arguments *args, uint64_t val, u32 val_len,
 			/* Make sure the string is null-terminated */
 			if (likely(len > 0))
 			{
+				if(len > max_arg_size)
+				{
+					len = max_arg_size;
```

Comment:
Let's double check one thing here:
https://linux.die.net/man/3/strlcpy

`strlcpy` returns the `strlen()` of the source string (does not include null terminator), while `max_arg_size` contains the maximum buffer length (including null terminator). Therefore here the `len` would be including the null terminator, while if the string was copied normally it wouldn't. I think at least, could you double check which is the case? Especially because the next check is using `len-1`.

Reply:
Ok I think I understand my confusion. Essentially, here we are using a `len` variable that could have two different meanings:
1. could be a number from 0 to `max_arg_size` (included) in case we used `strncpy_from_user`, indicating the length of the destination string including the null terminator, but excluding it if it is equal to `max_arg_size`
2. could be any integer in case we used `strlcpy` indicating length of the source string

However at the end of the function we want `len` to be _one_ thing which is the buffer length including the null terminator. The next lines of code work around this by not knowing what `len` means.

The logic sounds right in all cases. Honestly, I would propose to move this convoluted logic to the two branches like it was before some changes. It would be better if at the end of both the `strncpy` and `strlcpy` branch we had the same meaning for the value in `len`, so total buffer length incl. the terminator. The next line of code could simply be slapping the \0 so we're sure we are null terminating in all cases.

Reply:
there was already a bug in the `ppm_strncpy_from_user` before this fix https://github.com/falcosecurity/libs/pull/574 so I would go for the extra safe solution with the 

```c
		if(len > max_arg_size)
		{
			len = max_arg_size;
		}
```

shared by both kernel and user function